modules/swagger-codegen/src/main/java/com/wordnik/swagger/codegen/languages/JavaClientCodegen.java

```diff
@@ -71,9 +58,66 @@ public JavaClientCodegen() {
       );
     instantiationTypes.put("array", "ArrayList");
     instantiationTypes.put("map", "HashMap");
+    
+    cliOptions.add(new CliOption("invokerPackage", "root package for generated code"));
+    cliOptions.add(new CliOption("groupId", "groupId in generated pom.xml"));
+    cliOptions.add(new CliOption("artifactId", "artifactId in generated pom.xml"));
+    cliOptions.add(new CliOption("artifactVersion", "artifact version in generated pom.xml"));
+    cliOptions.add(new CliOption("sourceFolder", "source folder for generated code"));
   }
 
   @Override
+  public void processOpts() {
+    super.processOpts();
+    
+    if(additionalProperties.containsKey("invokerPackage")) {
+      this.setInvokerPackage((String)additionalProperties.get("invokerPackage"));
+    }
+    else{
+      //not set, use default to be passed to template
+      additionalProperties.put("invokerPackage", invokerPackage);
+    }
```

Comment:
Is it possible to introduce a general method of reading and storing the options from the provided JSON file, and place that method in e.g. `DefaultCodegen`? So that it doesn't need to do the same thing for every language separately.

A rough thought in my mind is using a `Map` to store any keys and values read from the JSON config file.